### ISSUE TITLE
tag skip_ansible_lint doesn't work on "when"

### ISSUE THREAD

**Issue by YogSottot** [ansible/ansible-lint#448]
# Issue Type
- Bug report

# Ansible and Ansible Lint details

```
ansible 2.7.5
  python version = 3.7.1 (default, Nov 23 2018, 10:01:49) [GCC 8.2.1 20181105 (Red Hat 8.2.1-5)]

ansible-lint 4.0.0
```

- ansible installation method: OS package
- ansible-lint installation method: pip

# Desired Behaviour

- Tasks with skip_ansible_lint are ignored
- The commented line is ignored

# Actual Behaviour (Bug report only)

### Sample of playbook.

```bash

- hosts: localhost
  tasks:
   - name: Update percona_mysql_percona_general_variables
     mysql_variables:
      variable={{ item.value.variable }}
      value={{ item.value.value }}
      login_unix_socket=/var/run/mysqld/mysqld.sock
     when: "percona_mysql_include_percona_specific_variables and item.value.include and item.value.dynamic and not percona_mysql_force_restart"
     with_dict: "{{ percona_mysql_general_variables }}"
     tags:
      - skip_ansible_lint

```
### Actual Behaviour  
```bash

[204] Lines should be no longer than 120 chars
playbook.yml:10
     when: "percona_mysql_include_percona_specific_variables and item.value.include and item.value.dynamic and not percona_mysql_force_restart"

```

In this example, we get  "[204]", even if the task has a skip_ansible_lint tag.  

### Sample of playbook2.

```bash

#- name: create error log file
#  file:
#    path={{ percona_mysql_log_dir }}/{{ percona_mysql_logdir_variables.percona_mysql_log_error.value }}
#    owner=mysql
#    group=root
#    mode=0640
#    seuser=system_u
#    serole=object_r
#    setype=mysqld_log_t
#    selevel=s0
#    state=touch
#  when: percona_mysql_error_log_check.stat.exists != true
#  tags:
#   - skip_ansible_lint

```
### Actual Behaviour
```bash

[601] Don't compare to literal True/False
playbook.yml:43
#  when: percona_mysql_error_log_check.stat.exists != true

```

And we get "[601]", even when this task is commented out with a "#".

**Comment by awcrosby** [comment]
For the 1st part: We will address the need to skip any rule in Issue #364 where we will add more granular skipping, and ensure it functions for any rule.
For the 2nd part: I added PR #453 to ignore comments for line-based rules.
